curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -209,10 +209,17 @@ private void processConfigData(byte[] data) throws Exception
         if (!properties.isEmpty())
         {
             QuorumMaj newConfig = new QuorumMaj(properties);
-            String connectionString = configToConnectionString(newConfig);
-            if (connectionString.trim().length() > 0)
+            String connectionString = configToConnectionString(newConfig).trim();
+            if (!connectionString.isEmpty())
             {
                 currentConfig.set(newConfig);
+                String oldConnectionString = ensembleProvider.getConnectionString();
+                int i = oldConnectionString.indexOf('/');
+                if (i >= 0)
```

Comment:
Perhap `== 0`? I'm not sure what if `i > 0` the string constructing.

Reply:
Hmm, we are extracting `chroot` from connection string. Everything last "/" (including "/" itself") are `chroot`.

The `if` part is similar to zookeeper.

https://github.com/apache/zookeeper/blob/a64dbf5b06ca1a73dc2ad6c7d31e679e312082aa/zookeeper-server/src/main/java/org/apache/zookeeper/client/ConnectStringParser.java#L55-L67